### PR TITLE
imagebuildah: drop the duplicate containerIDs list

### PR DESCRIPTION
**Pull request by nalind**
`StageExecutor`s keep their own lists of the IDs of containers that they create, so we don't need to merge them into a single list in the `Executor` that we'll then consult during clean up.  As long as we make sure we only ever try to delete a given stage's containers once, we can just use their lists directly, and avoid duplicating the information.

### CODE REVIEW

**Comment by rhatdan** [comment]
LGTM

**Comment by rhatdan** [comment]
@vrothberg @TomSweeneyRedHat PTAL

File: imagebuildah/build.go [hunk 1]
```diff
@@ -1569,15 +1566,23 @@ func processCopyFrom(dockerfiles []io.ReadCloser) []io.ReadCloser {
...
+			if err := b.store.DeleteContainer(ctr); err != nil {
+				logrus.Errorf("error deleting build container %q: %v\n", ctr, err)
+				lastErr = err
+			}
 		}
+		// The stages map includes some stages under multiple keys, so
+		// clearing their lists after we process a given stage is
+		// necessary to avoid triggering errors that would occur if we
+		// tried to delete a given stage's containers multiple times.
+		s.containerIDs = nil
```
Comment (vrothberg) [inline imagebuildah/build.go]: Doing this would also yield panics and reveal bugs in the code, which I think is a really good thing for such cases.

**Comment by nalind** [comment]
@rh-atomic-bot r=vrothberg

**Pull request closed without merging**